core/src/main/java/org/geysermc/geyser/session/cache/EntityCache.java

```diff
@@ -141,6 +142,16 @@ public Collection<PlayerEntity> getAllPlayerEntities() {
         return playerEntities.values();
     }
 
+    public void removeAllPlayerEntities() {
+        PlayerListPacket playerListPacket = new PlayerListPacket();
+        playerListPacket.setAction(PlayerListPacket.Action.REMOVE);
+        for (PlayerEntity otherEntity : playerEntities.values()) {
+            playerListPacket.getEntries().add(new PlayerListPacket.Entry(otherEntity.getTabListUuid()));
+        }
+        session.sendUpstreamPacket(playerListPacket);
+        playerEntities.clear();
+    }
```

Comment:
I wouldn't send the packet here, nothing else in the class does

Reply:
spawnEntity, removeEntity, removeAllEntities removeBossBar and removeAllBossBars does indirectly

Reply:
yeah you're right, sorry for glossing over the rest

i'm mostly just concerned with keeping the PlayerEntity stuff cohesive (having all or none of the add/remove methods deal with packet sending)